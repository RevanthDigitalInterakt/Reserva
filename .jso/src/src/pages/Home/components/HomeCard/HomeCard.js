  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _slicedToArray2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _react = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[2]));
  var _ImageComponent = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[3]));
  var _EventProvider = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[4]));
  var _configDeviceSizes = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[5]));
  var _testProps = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[6]));
  function HomeCard(_ref) {
    var imageUrl = _ref.imageUrl,
      reference = _ref.reference,
      reservaMini = _ref.reservaMini,
      orderBy = _ref.orderBy,
      filters = _ref.filters;
    var navigation = (0, _$$_REQUIRE(_dependencyMap[7]).useNavigation)();
    var handleNavigation = function handleNavigation() {
      var _filters$priceFilter, _filters$priceFilter2, _filters$priceFilter3;
      var _reference$split = reference.split(':'),
        _reference$split2 = (0, _slicedToArray2.default)(_reference$split, 2),
        categoryType = _reference$split2[0],
        categoryData = _reference$split2[1];
      if (categoryType === 'product') {
        _EventProvider.default.logEvent('page_view', {
          item_brand: _$$_REQUIRE(_dependencyMap[8]).defaultBrand.picapau
        });
        _EventProvider.default.logEvent('select_item', {
          item_list_id: categoryData || '',
          item_list_name: '',
          item_brand: _$$_REQUIRE(_dependencyMap[8]).defaultBrand.reserva
        });
        navigation.navigate('ProductDetail', {
          productId: categoryData,
          itemId: categoryData,
          colorSelected: '#FFFFFF'
        });
        return;
      }
      var navigateParams = {
        referenceId: reference,
        reservaMini: reservaMini,
        orderBy: orderBy
      };
      if ((filters != null && (_filters$priceFilter = filters.priceFilter) != null && _filters$priceFilter.from || (filters == null ? undefined : (_filters$priceFilter2 = filters.priceFilter) == null ? undefined : _filters$priceFilter2.from) === null) && filters != null && (_filters$priceFilter3 = filters.priceFilter) != null && _filters$priceFilter3.to) {
        var _filters$priceFilter4, _filters$priceFilter5;
        navigateParams = Object.assign({}, navigateParams, {
          filters: {
            priceFilter: {
              from: (filters == null ? undefined : (_filters$priceFilter4 = filters.priceFilter) == null ? undefined : _filters$priceFilter4.from) || 0,
              to: (filters == null ? undefined : (_filters$priceFilter5 = filters.priceFilter) == null ? undefined : _filters$priceFilter5.to) || 0
            }
          }
        });
      }
      navigation.navigate('ProductCatalog', navigateParams);
    };
    if (!imageUrl) return null;
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsx)(_$$_REQUIRE(_dependencyMap[10]).Box, Object.assign({}, (0, _testProps.default)(`card_container_${reference}`), {
      children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsx)(_$$_REQUIRE(_dependencyMap[11]).Button, Object.assign({
        onPress: handleNavigation
      }, (0, _testProps.default)(`card_button_${reference}`), {
        children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsx)(_ImageComponent.default, {
          height: 280,
          width: _configDeviceSizes.default.DEVICE_WIDTH * 0.85 - 16,
          source: {
            uri: imageUrl
          }
        })
      }))
    }));
  }
  var _default = exports.default = HomeCard;
