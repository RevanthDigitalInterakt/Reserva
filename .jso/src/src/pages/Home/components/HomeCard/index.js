  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function get() {
      return _HomeCard.default;
    }
  });
  var _HomeCard = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
