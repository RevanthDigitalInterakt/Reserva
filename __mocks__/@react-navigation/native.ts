const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
