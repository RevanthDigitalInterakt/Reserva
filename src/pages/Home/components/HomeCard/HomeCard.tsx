import React from 'react';
import { useNavigation } from '@react-navigation/native';
import { Box, Button } from '@usereservaapp/reserva-ui';
import EventProvider from '../../../../utils/EventProvider';
import { defaultBrand } from '../../../../utils/defaultWBrand';
import ImageComponent from '../../../../components/ImageComponent/ImageComponent';
import configDeviceSizes from '../../../../utils/configDeviceSizes';
import testProps from '../../../../utils/testProps';

export interface IHomeCard {
  imageUrl: string;
  reference: string;
  reservaMini: boolean;
  orderBy: string;
}

function HomeCard({
  imageUrl,
  reference,
  reservaMini,
  orderBy,
}: IHomeCard) {
  const navigation = useNavigation();

  const handleNavigation = () => {
    const [categoryType, categoryData] = reference.split(':');

    if (categoryType === 'product') {
      EventProvider.logEvent('page_view', {
        item_brand: defaultBrand.picapau,
      });

      EventProvider.logEvent('select_item', {
        item_list_id: categoryData || '',
        item_list_name: '',
        item_brand: defaultBrand.reserva,
      });

      navigation.navigate('ProductDetail', {
        productId: categoryData,
        itemId: categoryData,
        colorSelected: '#FFFFFF',
      });

      return;
    }

    navigation.navigate('ProductCatalog', {
      referenceId: reference,
      reservaMini,
      orderBy,
    });
  };

  if (!imageUrl) return null;

  return (
    <Box {...testProps(`card_container_${reference}`)}>
      <Button
        onPress={handleNavigation}
        {...testProps(`card_button_${reference}`)}
      >
        <ImageComponent
          height={280}
          width={configDeviceSizes.DEVICE_WIDTH * 0.85 - 16}
          source={{ uri: imageUrl }}
        />
      </Button>
    </Box>
  );
}

export default HomeCard;
