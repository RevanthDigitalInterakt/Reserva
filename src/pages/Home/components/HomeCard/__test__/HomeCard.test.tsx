import React from 'react';
import {
  fireEvent,
  render,
  screen,
  waitFor,
} from '@testing-library/react-native';
import { theme } from '@usereservaapp/reserva-ui';
import { ThemeProvider } from 'styled-components/native';
import HomeCard from '../HomeCard';

// MOCKS
const mockNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({ navigate: mockNavigate }),
}));

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <HomeCard
      reservaMini
      reference="category:1689"
      orderBy="RELEVANCIA"
      imageUrl="https://image.com"
    />
  </ThemeProvider>
);

describe('PrimeLP', () => {
  beforeEach(async () => {
    jest.useFakeTimers({ legacyFakeTimers: true });
    render(TestingComponent);
  });

  it('should render properly', () => {
    const card = screen.getByTestId('com.usereserva:id/card_container_category:1689');
    expect(card).toBeOnTheScreen();
  });

  it('should match to snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should call onAddPrimeToCart when to press call to action button', async () => {
    const callToAction = screen.getByTestId(
      'com.usereserva:id/card_button_category:1689',
    );

    fireEvent.press(callToAction);

    await waitFor(() => {
      expect(mockNavigate).toBeCalled();
      expect(mockNavigate).toBeCalledWith('ProductCatalog', {
        referenceId: 'category:1689',
        reservaMini: true,
        orderBy: 'RELEVANCIA',
      });
    });
  });
});
