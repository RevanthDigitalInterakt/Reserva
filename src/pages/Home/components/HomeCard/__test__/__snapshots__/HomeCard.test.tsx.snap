// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeLP should match to snapshot 1`] = `
<View
  accessibilityLabel="com.usereserva:id/card_container_category:1689"
  style={
    Array [
      Object {},
    ]
  }
  testID="com.usereserva:id/card_container_category:1689"
>
  <View
    alignItems="center"
    style={
      Array [
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <View
      accessibilityLabel="com.usereserva:id/card_button_category:1689"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="com.usereserva:id/card_button_category:1689"
    >
      <Image
        height={280}
        resizeMode="contain"
        source={
          Object {
            "uri": "https://image.com",
          }
        }
        style={
          Array [
            Object {
              "height": 0,
              "width": 621.5,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
